mlrun/api/initial_data.py

```diff
@@ -211,7 +211,13 @@ def _fix_datasets_large_previews(
                 artifact_dict
                 and artifact_dict.get("kind") == mlrun.artifacts.DatasetArtifact.kind
             ):
-                header = artifact_dict.get("header", [])
+                is_legacy = "metadata" not in artifact_dict
```

Comment:
Since I now see the heuristic is not only in the DB code, I guess it should sit in some common place, maybe static method of the base artifact class

Reply:
ok. Since it's used in multiple places (and in SQLDB it would be confusing to import `Artifact` since it's already using `Artifact` from the DB models package), I added a function in the mlrun helper utils.